cachito/web/models.py

```diff
@@ -590,10 +581,10 @@ def from_json(cls, kwargs):
                     # Send the changes queued up in SQLAlchemy to the database's transaction buffer.
                     # This will generate an ID that can be used below.
                     db.session.flush()
-                request_kwargs["user_id"] = submitted_for.id
-                request_kwargs["submitted_by_id"] = current_user.id
+                request_kwargs["user"] = submitted_for
+                request_kwargs["submitted_by"] = current_user
             else:
-                request_kwargs["user_id"] = current_user.id
+                request_kwargs["user"] = User.query.get(current_user.id)
```

Comment:
Could you please try the following instead to avoid an extra query?
```suggestion
                request_kwargs["user"] = current_user._get_current_object()
```

https://werkzeug.palletsprojects.com/en/1.0.x/local/#werkzeug.local.LocalProxy._get_current_object